examples/fluids/src/petsc_ops.c

```diff
@@ -499,4 +507,63 @@ PetscErrorCode CreateMatShell_Ceed(OperatorApplyContext ctx, Mat *mat) {
   PetscFunctionReturn(PETSC_SUCCESS);
 }
 
+/**
+ * @brief Return Mats for KSP solve
+ *
+ * Uses command-line flag with `ksp`'s prefix to determine if mat_ceed should be used directly or whether it should be assembled.
+ * If `Amat` is to be assembled, then `Pmat` is equal to `Amat`.
+ *
+ * If `Amat` uses `mat_ceed`, then `Pmat` is either assembled or uses `mat_ceed` based on the preconditioner choice in `ksp`.
+ *
+ * @param[in]  ksp      `KSP` object for used for solving
+ * @param[in]  mat_ceed `MATCEED` for the linear operator
+ * @param[out] Amat     `Mat` to be used for the solver `Amat`
+ * @param[out] Pmat     `Mat` to be used for the solver `Pmat`
+ * @param[in]  assemble Whether to assemble `Amat` and `Pmat` if they are not `mat_ceed`
+ */
+PetscErrorCode MatCeedReturnSolveOperators(KSP ksp, Mat mat_ceed, Mat *Amat, Mat *Pmat, PetscBool assemble) {
```

Comment:
The word `Return` in the code would imply that the return value is these 'solve operators'. I think you want `Get` or `Create`. Also, usually all in parameters come before all out parameters.

I don't think this is core `MatCEED` functionality, so the name is a bit confusing. A `KSP` is higher in the PETSc object hierarchy, so a `MatCEED` function shouldn't rely upon a `KSP`. This would be more of an internal application function, to my eye.

Reply:
Oh, agreed. I didn't think intend this to be a core `MatCEED` function. Though I agree, the name isn't very good.